### PR TITLE
feat: add model and execution provider-specific default batch sizes

### PR DESCRIPTION
**Pull request by tphakala**
## Summary

Implement intelligent default batch sizes based on model type and execution provider to optimize performance without requiring users to manually tune batch sizes for their hardware.

### Default Batch Sizes Matrix

| Model Type          | CPU | CUDA | TensorRT |
|---------------------|-----|------|----------|
| BirdNET 2.4         | 8   | 64   | 32       |
| BirdNET 3.0         | 8   | 32   | 32       |
| Perch v2            | 8   | 32   | 32       |
| BSG Finland         | 8   | 64   | 32       |

**Other GPU providers** (DirectML, ROCm, OpenVINO, CoreML, etc.): Conservative default of 16.

## Changes

- **Changed `DefaultsConfig.batch_size` to `Option<usize>`** to properly distinguish between "not set" vs "explicitly set to default value"
- **Added `determine_default_batch_size()` function** that uses actual execution provider (not requested device) to handle Auto/Gpu fallback scenarios correctly
- **Added CLI batch_size validation** to prevent `batch_size == 0` from bypassing config validation
- **User-specified batch sizes (CLI/config) always take precedence** over smart defaults
- **Added comprehensive tests** for all model/EP combinations including unknown providers

## Key Features

1. ✅ Uses **actual** EP (not predicted) - handles all Auto/Gpu fallback scenarios
2. ✅ Respects explicit user choices (CLI `-b` flag or config file)
3. ✅ Minimal code changes (single insertion point after classifier build)
4. ✅ Works with library detection and CPU fallbacks
5. ✅ Conservative defaults for unknown/future providers

## Testing

- All existing tests pass
- New unit tests cover:
  - All model types with CPU
  - All model types with CUDA
  - All model types with TensorRT
  - Other GPU providers (DirectML, etc.)
  - CPU fallback scenarios
  - Unknown provider fallback

## Code Review

Reviewed and approved by both Claude Sonnet 4.5 and Gemini 3 Pro:
- ✅ Logic correctness verified
- ✅ Edge cases handled (config vs CLI precedence, fallbacks)
- ✅ No race conditions (sequential execution)
- ✅ Proper validation (config + CLI)
- ✅ Comprehensive test coverage

## Breaking Changes

None. Existing configurations continue to work:
- Users with explicit `batch_size` in config: value is respected
- Users with no `batch_size` setting: get smart defaults (previously got 8)
- Users with CLI `-b` flag: value is respected

## Example Usage

```bash
# Uses smart default (e.g., 64 for BirdNET 2.4 + CUDA)
birda --cuda -m birdnet-v24 audio.wav

# Explicit batch size always takes precedence
birda --cuda -m birdnet-v24 -b 128 audio.wav

# CPU gets batch size 8 (optimized for CPU)
birda --cpu audio.wav
```

## Release Notes

* **New Features**
  * Batch size configuration is now optional. The system automatically selects intelligent defaults based on your model type and execution provider, while still allowing manual override if needed.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -640,6 +687,28 @@ fn analyze_files(
...
+        );
+        default
+    });
+
+    // Validate batch size (catch CLI -b 0 case)
+    if batch_size == 0 {
+        return Err(Error::ConfigValidation {
+            message: "batch_size must be at least 1".to_string(),
+        });
+    }
```
[the remark this answers is not in the document]
Comment (tphakala) [reply src/lib.rs]: Fixed in fe1f401. Moved validation to argument parsing level using clap's value_parser with parse_batch_size() validator. Now catches batch_size < 1 at CLI parsing time, before any model loading. Config validation still handles batch_size = 0 in TOML files.
